statsd/reporter.go

```diff
@@ -148,9 +147,17 @@ func (r *cactusStatsReporter) Reporting() bool {
 }
 
 func (r *cactusStatsReporter) Tagging() bool {
-	return false
+	return true
 }
 
 func (r *cactusStatsReporter) Flush() {
 	// no-op
 }
+
+func getStatsdTagPairs(tags map[string]string) []statsd.Tag {
```

Comment:
- Keeping it as a func and not a method. 
![image](https://user-images.githubusercontent.com/12872673/125498798-4f107f8f-761c-4258-be13-497306255301.png)


- 1 allocation here when the array moves to heap.

```
~/gocode/src/github.com/uber-go/tally/statsd master !4 ?1 ❯ go test . -run none -bench BenchmarkGetStatsdTagPairs --benchtime 3s -benchmem      ▼
goos: darwin
goarch: amd64
pkg: github.com/uber-go/tally/statsd
cpu: Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz
BenchmarkGetStatsdTagPairs-12    	31455973	       105.0 ns/op	      64 B/op	       1 allocs/op
PASS
ok  	github.com/uber-go/tally/statsd	3.943s
```

- Benchmark 
```

func BenchmarkGetStatsdTagPairs(b *testing.B) {
	tags := map[string]string{
		"K1": "V1",
		"K2": "V2",
	}
	b.ResetTimer()
	for i := 0; i < b.N; i++ {
		_ = getStatsdTagPairs(tags)
	}
}
```